subsys/net/ip/ipv4.c

```diff
@@ -330,6 +330,11 @@ enum net_verdict net_ipv4_input(struct net_pkt *pkt)
 
 	net_pkt_set_family(pkt, PF_INET);
 
+	if (!net_pkt_filter_ipv4_recv_ok(pkt)) {
```

Comment:
Is the location of this hook correct? For IPv6 we call the hook before verifying the destination IP address, here we call it after.

Reply:
Moved to more appropriate place